repos/system_upgrade/common/libraries/tests/test_repofileutils.py

```diff
@@ -12,6 +15,23 @@ def test_invert_dict():
     assert inv_dict == {'a': [1], 'b': [1, 2]}
 
 
+def test__parse_repository_missing_name():
+    repoid = 'missing-name'
+    data = {'baseurl': 'http://example.com', 'enabled': '1', 'gpgcheck': '1'}
```

Comment:
I think it wouldn't hurt to add some more test cases.

Since you test the `_parse_repository_function` directly, you have more flexibility to test unexpected inputs. 
I would for example add a test with `'name'=None`, some test to make sure invalid type is caught (e.g. `'baseurl'=True`) and also a test with `repoid=None`. 

The inputs right now should always be a string, but this will catch errors if we change the function in the future or call it from somewhere else.

Reply:
I added those test cases.